### PR TITLE
[Feature-14545][Master][UI]Cross workflow parameter passing

### LINKED ISSUE #14545: [Feature][Master][UI] Cross-workflow parameter passing
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

In the subprocess task, the subworkflow can choose its own output parameters to output to the downstream tasks in the parent workflow.
This is a [DolphinScheduler project of OSPP-2023.](https://summer-ospp.ac.cn/org/prodetail/232290357)

### Use case

Example:
This is the output parameter of the task in the subworkflow
image
This is the output parameter defined in the subworkflow
image
Output parameter1 and parameter2 in the downstream task of the subworkflow task, and finally output parameter1 successfully
image

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Orange-Summer**
## Purpose of the pull request

cross-workflow parameter passing
close #14545 

## Brief change log

add relevant code in module dolphinscheduler-ui
add code in `submitPostNode` method of `WorkflowExecuteRunnable.java`
add method `crossWorkflowParameterPassing` in `WorkflowExecuteThreadPool.java`
Design and usage documentation can be seen here #14545 

## Verify this pull request

This change can be verified as follows:

Manually verified the change by testing locally.

### CODE REVIEW

**Review by SbloodyS — COMMENTED** [review]
Please update the docs too.

**Review by songjianet — DISMISSED** [review]

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java [hunk 1]
```diff
@@ -182,6 +185,18 @@ private void notifyProcessChanged(ProcessInstance finishProcessInstance) {
...
 
+    private void crossWorkflowParameterPassing(ProcessInstance finishProcessInstance, TaskInstance taskInstance) {
+        try {
+            MasterTaskExecuteRunnable masterTaskExecuteRunnable =
+                    MasterTaskExecuteRunnableHolder.getMasterTaskExecuteRunnable(taskInstance.getId());
+            masterTaskExecuteRunnable.getILogicTask().getTaskParameters()
+                    .setVarPool(finishProcessInstance.getVarPool());
+            log.info("Cross workflow parameter passing success");
+        } catch (Exception ex) {
+            log.info("Cross workflow parameter passing error");
```
Comment (SbloodyS) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java]:
```suggestion
            log.error("Cross workflow parameter passing error: {}", ex);
```

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java [hunk 2]
```diff
@@ -182,6 +185,18 @@ private void notifyProcessChanged(ProcessInstance finishProcessInstance) {
         }
     }
 
+    private void crossWorkflowParameterPassing(ProcessInstance finishProcessInstance, TaskInstance taskInstance) {
+        try {
+            MasterTaskExecuteRunnable masterTaskExecuteRunnable =
+                    MasterTaskExecuteRunnableHolder.getMasterTaskExecuteRunnable(taskInstance.getId());
+            masterTaskExecuteRunnable.getILogicTask().getTaskParameters()
+                    .setVarPool(finishProcessInstance.getVarPool());
+            log.info("Cross workflow parameter passing success");
```
Comment (SbloodyS) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java]:
```suggestion
            log.info("Cross workflow parameter passing success, processInstanceId: {}, taskInstanceId");
```

It's best to add more effective information for troubleshooting purposes.

**Review by SbloodyS — CHANGES_REQUESTED** [review]

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java [hunk 3]
```diff
@@ -182,6 +185,18 @@ private void notifyProcessChanged(ProcessInstance finishProcessInstance) {
         }
     }
 
+    private void crossWorkflowParameterPassing(ProcessInstance finishProcessInstance, TaskInstance taskInstance) {
+        try {
+            MasterTaskExecuteRunnable masterTaskExecuteRunnable =
+                    MasterTaskExecuteRunnableHolder.getMasterTaskExecuteRunnable(taskInstance.getId());
+            masterTaskExecuteRunnable.getILogicTask().getTaskParameters()
+                    .setVarPool(finishProcessInstance.getVarPool());
+            log.info("Cross workflow parameter passing success, processInstanceId: {}, taskInstanceId");
```
Comment (SbloodyS) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java]: The processInstanceId and taskInstanceId Not filled.

**Review by SbloodyS — CHANGES_REQUESTED** [review]

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java (hunk 3 above)
Comment (Orange-Summer) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java]: Sorry, I didn't check carefully
`log.info("Cross workflow parameter passing success, finishProcessInstanceId: {}, taskInstanceId: {}", finishProcessInstance.getId(), taskInstance.getId());`
is this ok?

Comment (SbloodyS) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java]: Yes. Please supplement other relevant logs.

Comment (Orange-Summer) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java]: Is there anything else you think needs to be added to the log?

Comment (SbloodyS) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java]: Exception error log should add too.

Comment (Orange-Summer) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java]: I have added it, thanks for your advice.

**Review by SbloodyS — APPROVED** [review]

**Pull request merged**